pageserver/src/tenant/timeline.rs

```diff
@@ -4077,7 +4080,18 @@ impl Timeline {
 
         // Backpressure mechanism: wait with continuation of the flush loop until we have uploaded all layer files.
         // This makes us refuse ingest until the new layers have been persisted to the remote.
-        self.remote_client.wait_completion().await;
+        self.remote_client
+            .wait_completion()
+            .await
+            .map_err(|e| match e {
+                WaitCompletionError::UploadQueueShutDownOrStopped
+                | WaitCompletionError::NotInitialized(
+                    NotInitialized::ShuttingDown | NotInitialized::Stopped,
+                ) => FlushLayerError::Cancelled,
+                WaitCompletionError::NotInitialized(NotInitialized::Uninitialized) => {
+                    FlushLayerError::Other(anyhow!(e).into())
+                }
+            })?;
```

Comment:
I would had thought there was a conversion `From<WaitCompletionError> for FlushLayerError`? maybe it was another map_err?

Reply:
There is no conversion, if I remove the map_err it gives an error.